src/DevChatter.Bot/Startup/SetUpBot.cs

```diff
@@ -25,48 +23,49 @@ public static IContainer NewBotDepedencyContainer(BotConfiguration botConfigurat
 
             var builder = new ContainerBuilder();
 
-            builder.Register(ctx => botConfiguration.CommandHandlerSettings).AsSelf().InstancePerLifetimeScope();
```

Comment:
@benrick I'm highly confident that `InstancePerLifetimeScope()` should be called here, because we get a new lifetime scope whenever the restart command is used in `Program.WaitForCommands(IContainer container)`